File: scripts/import_correct_steps.py
#!/usr/bin/env python3
"""
从 exercises_v2 原始数据导入 correct_steps 和 correct_steps_zh 到数据库
"""
import json
import os
import mysql.connector
from pathlib import Path
import sys

# MySQL 配置
MYSQL_CONFIG = {
    'host': os.getenv('MYSQL_HOST', 'localhost'),
    'user': os.getenv('MYSQL_USER', 'fitness_user'),
    'password': os.getenv('MYSQL_PASSWORD', 'fitness_pass'),
    'database': os.getenv('MYSQL_DATABASE', 'fitness_app'),
    'port': int(os.getenv('MYSQL_PORT', 3306))
}

# 原始数据目录
EXERCISES_V2_PATH = Path(__file__).parent.parent / 'storage' / 'app' / 'public' / 'exercises_v2'

def connect_db():
    """连接到 MySQL 数据库"""
    try:
        conn = mysql.connector.connect(**MYSQL_CONFIG)
        return conn
    except mysql.connector.Error as err:
        print(f"❌ 数据库连接失败: {err}")
        sys.exit(1)

def load_exercise_data(json_path):
    """加载 JSON 文件中的动作数据"""
    try:
        with open(json_path, 'r', encoding='utf-8') as f:
            return json.load(f)
    except Exception as e:
        print(f"❌ 加载 JSON 失败 {json_path}: {e}")
        return None

def import_correct_steps():
    """导入 correct_steps 数据"""
    conn = connect_db()
    cursor = conn.cursor()
    
    imported_count = 0
    skipped_count = 0
    error_count = 0
    
    print(f"🔍 扫描数据目录: {EXERCISES_V2_PATH}")
    
    # 递归扫描所有 data.json 文件
    for json_file in EXERCISES_V2_PATH.rglob('data.json'):
        try:
            data = load_exercise_data(json_file)
            if not data:
                continue
            
            exercise_id = data.get('id')
            if not exercise_id:
                skipped_count += 1
                continue
            
            # 获取 correct_steps 和 correct_steps_zh
            correct_steps = data.get('correct_steps')
            correct_steps_zh = data.get('correct_steps_zh')
            
            # 如果没有 correct_steps，尝试从 description 中提取（如果和 description 重复就说明没有数据）
            if not correct_steps and not correct_steps_zh:
                # 检查 description 是否包含步骤信息
                description = data.get('description', '')
                if 'Correct Steps' in description or 'correct' in description.lower():
                    print(f"⚠️  Exercise {exercise_id}: description 包含步骤，但 correct_steps 为空（可能已复制）")
                skipped_count += 1
                continue
            
            # 转换为 JSON 字符串
            correct_steps_json = json.dumps(correct_steps) if correct_steps else None
            correct_steps_zh_json = json.dumps(correct_steps_zh) if correct_steps_zh else None
            
            # 更新数据库
            update_query = """
                UPDATE exercises 
                SET correct_steps = %s, correct_steps_zh = %s
                WHERE id = %s
            """
            
            cursor.execute(update_query, (correct_steps_json, correct_steps_zh_json, exercise_id))
            conn.commit()
            
            imported_count += 1
            if imported_count % 10 == 0:
                print(f"✅ 已导入 {imported_count} 条数据...")
        
        except Exception as e:
            print(f"❌ 处理失败 {json_file}: {e}")
            error_count += 1
            continue
    
    cursor.close()
    conn.close()
    
    # 输出统计
    print("\n" + "="*50)
    print(f"📊 导入统计结果:")
    print(f"✅ 成功导入: {imported_count} 条")
    print(f"⏭️  跳过: {skipped_count} 条")
    print(f"❌ 错误: {error_count} 条")
    print("="*50)

if __name__ == '__main__':
    print("🚀 开始导入 correct_steps 数据...")
    import_correct_steps()
    print("✅ 导入完成！")

File: scripts/import_correct_steps.php
<?php
/**
 * 从 exercises_v2 原始数据导入 correct_steps 到数据库
 * 使用方式: php import_correct_steps.php
 */

require_once __DIR__ . '/../bootstrap/app.php';

use Illuminate\Support\Facades\DB;
use Illuminate\Support\Facades\File;
use Illuminate\Support\Collection;

class ExerciseStepsImporter
{
    private string $dataPath;
    private int $importedCount = 0;
    private int $skippedCount = 0;
    private int $errorCount = 0;

    public function __construct()
    {
        $this->dataPath = storage_path('app/public/exercises_v2');
    }

    /**
     * 开始导入
     */
    public function import(): void
    {
        echo "🚀 开始导入 correct_steps 数据...\n";
        echo "📂 数据目录: {$this->dataPath}\n\n";

        $files = $this->findJsonFiles();
        echo "🔍 找到 " . count($files) . " 个 data.json 文件\n\n";

        foreach ($files as $jsonFile) {
            $this->processFile($jsonFile);
        }

        $this->printStatistics();
    }

    /**
     * 递归查找所有 data.json 文件
     */
    private function findJsonFiles(): array
    {
        $files = [];
        $iterator = new \RecursiveIteratorIterator(
            new \RecursiveDirectoryIterator($this->dataPath)
        );

        foreach ($iterator as $file) {
            if ($file->getFilename() === 'data.json') {
                $files[] = $file->getRealPath();
            }
        }

        return $files;
    }

    /**
     * 处理单个 JSON 文件
     */
    private function processFile(string $jsonFile): void
    {
        try {
            $content = file_get_contents($jsonFile);
            $data = json_decode($content, true, 512, JSON_UNESCAPED_UNICODE);

            if (!$data || !isset($data['id'])) {
                $this->skippedCount++;
                return;
            }

            $exerciseId = $data['id'];
            $correctSteps = $data['correct_steps'] ?? null;
            $correctStepsZh = $data['correct_steps_zh'] ?? null;

            // 如果两个字段都为空，检查是否复制了 description
            if (!$correctSteps && !$correctStepsZh) {
                $description = $data['description'] ?? '';
                if (str_contains($description, 'Correct Steps') || str_contains(strtolower($description), 'correct')) {
                    echo "⚠️  Exercise {$exerciseId}: description 包含步骤，但 correct_steps 为空\n";
                }
                $this->skippedCount++;
                return;
            }

            // 更新数据库
            DB::table('exercises')
                ->where('id', $exerciseId)
                ->update([
                    'correct_steps' => $correctSteps ? json_encode($correctSteps, JSON_UNESCAPED_UNICODE) : null,
                    'correct_steps_zh' => $correctStepsZh ? json_encode($correctStepsZh, JSON_UNESCAPED_UNICODE) : null,
                ]);

            $this->importedCount++;

            if ($this->importedCount % 10 === 0) {
                echo "✅ 已导入 {$this->importedCount} 条数据...\n";
            }

        } catch (\Exception $e) {
            echo "❌ 处理失败 {$jsonFile}: {$e->getMessage()}\n";
            $this->errorCount++;
        }
    }

    /**
     * 输出统计结果
     */
    private function printStatistics(): void
    {
        echo "\n" . str_repeat("=", 50) . "\n";
        echo "📊 导入统计结果:\n";
        echo "✅ 成功导入: {$this->importedCount} 条\n";
        echo "⏭️  跳过: {$this->skippedCount} 条\n";
        echo "❌ 错误: {$this->errorCount} 条\n";
        echo str_repeat("=", 50) . "\n";
    }
}

// 执行导入
$importer = new ExerciseStepsImporter();
$importer->import();
echo "✅ 导入完成！\n";
